### PR TITLE
jetpack-mu-wpcom: remove linting rule overrides

### PR DESCRIPTION
**Pull request by tbradsha**
This corrects some lints and then removes the following overrides in said package:

```
'jsdoc/require-returns': 'off',
'jsdoc/require-param-type': 'off',
```

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Is CI happy?
* Does this run fine? `pnpm lint-file projects/packages/jetpack-mu-wpcom/`

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/eslint.config.mjs [hunk 1]
```diff
@@ -7,9 +7,6 @@ export default [
 		rules: {
 			'testing-library/prefer-screen-queries': 'off',
 			'react/jsx-no-bind': 'off',
-			// @todo Turn these back on for JS here. The block below handles TS.
-			'jsdoc/require-returns': 'off',
```
Comment (anomiex) [inline projects/packages/jetpack-mu-wpcom/eslint.config.mjs:11]: Looks like I had a typo here, that should have been `jsdoc/require-returns-type` like below and in [the old code](https://github.com/Automattic/jetpack/blob/44813737107aa90c148fc9a6a185d12e02b6d06d/projects/packages/jetpack-mu-wpcom/.eslintrc.js#L15). 😳

File: projects/packages/jetpack-mu-wpcom/src/common/tour-kit/components/keyboard-navigation.tsx [hunk 2]
```diff
@@ -28,6 +28,7 @@ const KeyboardNavigation: React.FunctionComponent< Props > = ( {
 } ) => {
 	/**
 	 * Expand Tour Nav
+	 * @return {null} This component is non-rendering.
```
Comment (anomiex) [inline projects/packages/jetpack-mu-wpcom/src/common/tour-kit/components/keyboard-navigation.tsx:31]: I verified that in TS files it's also ok with something like this
```
	 * @return This component is non-rendering.
```
I have no opinion on whether you should make that sort of change here and in other TS files in this PR. 🙂

**Review by anomiex — APPROVED** [review]
The added docs seem plausible, but I didn't dig in to verify them.

File: projects/packages/jetpack-mu-wpcom/src/common/tour-kit/components/keyboard-navigation.tsx (hunk 2 above)
Comment (tbradsha) [reply projects/packages/jetpack-mu-wpcom/src/common/tour-kit/components/keyboard-navigation.tsx:31]: No opinion myself either, but the extra doc shouldn't hurt here so I'll leave it.

**Pull request merged**